### PR TITLE
Episode nav

### PR DESCRIPTION
**Pull request by fshenton**
Unclear why this pull request contains your Link refactor and my merges to markup?

The episode-nav branch should contain only the EpisodesControls changes as that's the only thing I committed to it. :worried:

### CODE REVIEW

**Comment by pookage** [comment]
> Unclear why this pull request contains your Link refactor and my merges to markup?
> 
> The episode-nav branch should contain only the EpisodesControls changes as that's the only thing I committed to it. 😟

So, this will be because `episode-context` will have been branched from `markup` before my `link-refactor` branch was merged in; therefore that commit is new to this branch. It has the same commit hash as the one on `markup`, though, so those two commits won't double-up when this is merged into `develop`, don't worry 👌

File: src/components/Episodes/Episodes.jsx [hunk 1]
```diff
@@ -8,23 +9,20 @@ import {
...
+				<EpisodesProvider>
 					<ol>
-						{ pagination }
+						{ episodes }
 					</ol>
-				</nav>
+				</EpisodesProvider>
+			</article>
+			<EpisodesProvider>
+				<EpisodesControls />
```
Comment (pookage) [inline src/components/Episodes/Episodes.jsx]: SUPER minor nitpick - could we name this component `<EpisodesNav />` instead? 🙏

File: src/components/Episodes/Episodes.jsx [hunk 2]
```diff
@@ -8,23 +9,20 @@ import {
...
-			<EpisodesProvider>
-				<ol>
-					{ episodes }
-				</ol>
-				<nav>
-					<button aria-label="Previous episode." />
-					<button aria-label="Next episode." />
+		<>
+			<article>
+				<EpisodesProvider>
```
Comment (pookage) [inline src/components/Episodes/Episodes.jsx]: SO, we need `<EpisodesProvider>` to wrap _both_ the `episodes` _and_ the `<EpisodesControls>`, if we don't then we're creating two instances of the component, and thus there'll be two separate states with `activeIndex` being provided.

File: src/components/EpisodesControls/EpisodesControls.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, { useContext } from "react";
+import { data } from "COMPONENTS/Episodes/";
+import { RENDER } from "./";
+import { Episodes } from "COMPONENTS/Episodes/";
```
Comment (pookage) [inline src/components/EpisodesControls/EpisodesControls.jsx]: this should be imported from line 2 as you're already importing something from that component above...

File: src/components/EpisodesControls/EpisodesControls.jsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	const { activeIndex } = useContext(Episodes);
+
+	//RENDER
+	//---------------------------
+	const pagination = data.map(RENDER.pagination);
+
+	return (
+		<nav>
+			<button aria-label="Previous episode." />
+			<button aria-label="Next episode." />
```
Comment (pookage) [inline src/components/EpisodesControls/EpisodesControls.jsx]: in addition to my comment from the other branch (these buttons should have `aria-controls` for the `<ol>` of `episodes`, it occurrs to me that they also control the pagination below as well (ie. which one has `aria-current`)

File: src/components/EpisodesControls/renderUtils.jsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import { s } from "./";
+import Link from "COMPONENTS/Link/";
+
+function renderPaginationLink(data){
+
+	const { number } = data;
+
+	return (
+		<li key={`episode__pagination__${number}`}>
```
Comment (pookage) [inline src/components/EpisodesControls/renderUtils.jsx]: I've just found out that [aria-current](https://www.w3.org/TR/wai-aria-1.1/#aria-current) exists! can we use that here? If we grab the `activeIndex` from the provider and compare it against the `index` here, then we should be able to apply that.

File: src/components/Episodes/Episodes.jsx (hunk 1 above)
Comment (fshenton) [reply src/components/Episodes/Episodes.jsx]: Yarp. done

File: src/components/Episodes/Episodes.jsx (hunk 2 above)
Comment (fshenton) [reply src/components/Episodes/Episodes.jsx]: Cool, had a feeling that would come back to me!
Wasn't sure about the nav being in the article, but it definitely seemed weird as sibling when it meant adding provider again.

File: src/components/EpisodesControls/EpisodesControls.jsx (hunk 3 above)
Comment (fshenton) [reply src/components/EpisodesControls/EpisodesControls.jsx]: Done, however Episodes has been renamed in the context branch, so i'll want to update that. Sent you a PM about this as unsure the best way to sync that rename from context.

Comment (fshenton) [reply src/components/EpisodesControls/EpisodesControls.jsx]: Sorted.

File: src/components/EpisodesControls/EpisodesControls.jsx (hunk 4 above)
Comment (fshenton) [reply src/components/EpisodesControls/EpisodesControls.jsx]: Added controls for episodes list.
TODO: using context to set aria-current and then looking at controls with that also.

File: src/components/EpisodesControls/renderUtils.jsx (hunk 5 above)
Comment (fshenton) [reply src/components/EpisodesControls/renderUtils.jsx]: Now importing context and setting aria-current based on activeindex.

File: src/components/EpisodesControls/EpisodesControls.jsx (hunk 4 above)
Comment (fshenton) [reply src/components/EpisodesControls/EpisodesControls.jsx]: Controls now point to the episode list and the pagination list.
Aria-current determined by activeindex.

File: src/components/Episodes/Episodes.jsx [hunk 6]
```diff
@@ -8,32 +9,21 @@ import {
 
 export default function Episodes(){
 
-	const pagination = data.map(RENDER.pagination);
 	const episodes = data.map(RENDER.episode);
 
 	return (
+		<>
```
Comment (pookage) [inline src/components/Episodes/Episodes.jsx]: don't need a fragment if there's already a wrapper 👌

Comment (fshenton) [reply src/components/Episodes/Episodes.jsx]: Removed fragment!

**Pull request merged**